packages/host/app/components/matrix/forgot-password.gts

```diff
@@ -0,0 +1,447 @@
+        width: 100%;
+      }
+      .or {
+        margin: var(--boxel-sp-sm);
+        font: 500 var(--boxel-font-sm);
+      }
+      .button {
+        --boxel-button-padding: var(--boxel-sp-sm);
+        width: fit-content;
+        margin-top: var(--boxel-sp-lg);
+      }
+      .button :deep(.boxel-loading-indicator) {
+        display: flex;
+        justify-content: center;
+        align-items: center;
+      }
+      .email-validation-instruction {
+        padding: 0;
+        list-style-position: inside;
+        margin-top: 0;
+        margin-bottom: var(--boxel-sp);
+      }
+      .email-validation-instruction li {
+        margin-bottom: var(--boxel-sp-sm);
+      }
+      .error {
+        color: var(--boxel-error-100);
+        padding: 0;
+        font: 500 var(--boxel-font-xs);
+        margin: var(--boxel-sp-xxs) auto 0 auto;
+      }
+    </style>
+  </template>
+
+  @tracked private error: string | undefined;
+  @tracked private email: string | undefined;
+  @tracked private emailError: string | undefined;
+  @tracked private password: string | undefined;
+  @tracked private confirmPassword: string | undefined;
+  @tracked private passwordError: string | undefined;
+  @tracked private confirmPasswordError: string | undefined;
+  @tracked private state:
+    | { type: 'initial' }
+    | { type: 'validateEmail'; email: string; sendAttempt: number }
+    | {
+        type: 'waitForEmailValidation';
+        email: string;
+        sendAttempt: number;
+        clientSecret: string;
+        sid: string;
+      }
+    | {
+        type: 'resetPassword';
+        email: string;
+        sendAttempt: number;
+        clientSecret: string;
+        sid: string;
+      }
+    | { type: 'resetPasswordSuccess' } = {
+    type: 'initial',
+  };
+  @service private declare matrixService: MatrixService;
+
+  private get isForgotPasswordBtnDisabled() {
+    return (
+      !this.email || this.emailError || this.sendEmailValidationTask.isRunning
+    );
+  }
+
+  private get isResetPasswordBtnDisabled() {
+    return (
+      !this.password ||
+      !this.confirmPassword ||
+      this.passwordError ||
+      this.confirmPasswordError ||
+      this.resetPassword.isRunning
+    );
+  }
+
+  @action
+  private setEmail(email: string) {
+    this.email = email;
+    this.emailError = undefined;
+  }
+
+  @action
+  private setPassword(password: string) {
+    this.password = password;
+    this.passwordError = undefined;
+    this.error = undefined;
+  }
+
+  @action
+  private setConfirmPassword(confirmPassword: string) {
+    this.confirmPassword = confirmPassword;
+    this.confirmPasswordError = undefined;
+    this.error = undefined;
+  }
+
+  @action
+  private checkPassword() {
+    if (!this.password) {
+      this.passwordError = 'Password is missing';
+    } else if (isValidPassword(this.password)) {
+      this.passwordError =
+        'Password must be at least 8 characters long and include a number and a symbol';
```

Comment:
Maybe we can just let the user pick whatever they want. Personally I find these rules pretty annoying. Or is this a matrix requirement?

Reply:
That requirement is from this ticket https://linear.app/cardstack/issue/CS-6138/allow-account-creation-via-sign-up-screen.